crates/modelardb_storage/src/test.rs

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2023 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Implementation of storage related functions and constants used throughout ModelarDB's tests.
+
+use std::sync::Arc;
+
+use arrow::array::{BinaryArray, Float32Array, RecordBatch, UInt16Array, UInt64Array, UInt8Array};
+use arrow::datatypes::{ArrowPrimitiveType, DataType, Field, Schema};
+use modelardb_common::test::{ERROR_BOUND_FIVE, ERROR_BOUND_ONE, ERROR_BOUND_ZERO};
+use modelardb_types::functions;
+use modelardb_types::schemas::COMPRESSED_SCHEMA;
+use modelardb_types::types::{ArrowTimestamp, ArrowValue, ErrorBound, TimestampArray, ValueArray};
+
+use crate::metadata::model_table_metadata::ModelTableMetadata;
+
+/// SQL to create a normal table with a timestamp column and two floating point columns.
+pub const NORMAL_TABLE_SQL: &str =
```

Comment:
As more and more test functionality is created it is becoming hard to keep track of where it is located, thus possible leading to code duplication. So should we make a `modelardb_test` crate with test functionality organized into modules (e.g., schemas, data generation, normal table, model table etc.) that can be used as a development dependency in the other crates? If we only import it as a development dependency it should be possible to not have the crates that depend on it included it in debug and release builds, thus making the dependencies used in `modelardb_test` less of a concern. If we want to do this, it should probably be done in a separate PR, so maybe create an issue for now.

Reply:
Good idea, I have created an issue for this (https://github.com/ModelarData/ModelarDB-RS/issues/260) that links back to this comment.